File: cblue-ai/src/common/pii_redaction.ts
export class PIIRedactor {
    private static PATTERNS: Record<string, RegExp> = {
        phone: /\b0[0-9]{1,2}[-\s]?[0-9]{3}[-\s]?[0-9]{4}\b/g,
        email: /\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b/g,
        thai_id: /\b[0-9]{1}-[0-9]{4}-[0-9]{5}-[0-9]{2}-[0-9]\b/g,
    };

    static redact(text: string): string {
        let redactedText = text;
        for (const [name, pattern] of Object.entries(PIIRedactor.PATTERNS)) {
            redactedText = redactedText.replace(pattern, `[REDACTED_${name.toUpperCase()}]`);
        }
        return redactedText;
    }

    static detect(text: string): boolean {
        return Object.values(PIIRedactor.PATTERNS).some(pattern => pattern.test(text));
    }
}

File: cblue-ai/common/pii_redaction.py
import re

class PIIRedactor:
    PATTERNS = {
        'phone': r'\b0[0-9]{1,2}[-\s]?[0-9]{3}[-\s]?[0-9]{4}\b',
        'email': r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b',
        'thai_id': r'\b[0-9]{1}-[0-9]{4}-[0-9]{5}-[0-9]{2}-[0-9]\b',
    }
    
    @staticmethod
    def redact(text: str) -> str:
        for name, pattern in PIIRedactor.PATTERNS.items():
            text = re.sub(pattern, f'[REDACTED_{name.upper()}]', text)
        return text
    
    @staticmethod
    def detect(text: str) -> bool:
        return any(re.search(pattern, text) for pattern in PIIRedactor.PATTERNS.values())
